pkg/util/deployment/deployment.go

```diff
@@ -40,6 +45,32 @@ const (
 	RollbackDone              = "DeploymentRollback"
 )
 
+// IsScalingEvent checks whether the provided deployment has been updated with a scaling event.
+func IsScalingEvent(d extensions.Deployment) bool {
+	surgeAnnotation, ok := d.Annotations[SurgePodsAnnotation]
+	if !ok {
+		// We are not sure so we should fallback to the rollout logic
+		return false
+	}
+	surge, err := strconv.Atoi(surgeAnnotation)
+	if err != nil {
+		// Messed-up annotation; fallback to the rollout logic
+		return false
+	}
+	maybeMaxSurge := 0
+	if d.Spec.Strategy.Type == extensions.RollingUpdateDeploymentStrategyType {
+		maybeMaxSurge, isPercent, err := util.GetIntOrPercentValue(&d.Spec.Strategy.RollingUpdate.MaxSurge)
+		if err != nil {
+			// Messed-up maxSurge field; fallback to the rollout logic
+			return false
+		}
+		if isPercent {
+			maybeMaxSurge = util.GetValueFromPercent(maybeMaxSurge, d.Spec.Replicas)
+		}
+	}
+	return d.Spec.Replicas+maybeMaxSurge != surge
```

Comment:
This is what this code is implementing. If any of those changes, then the sum will change, then there will be a mismatch between the sum and the previous sum (annotation), then we know it's a scaling event:)